lib/exercises.rb

```diff
@@ -2,29 +2,179 @@
 
 # This method will return an array of arrays.
 # Each subarray will have strings which are anagrams of each other
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n) where n is the number of strings
+# Space Complexity: O(n) where n is the number of strings
+def grouped_anagrams_version_1(strings)
+  length = strings.length
+  # kind of arbitrary but it seems big enough to get a good spread?
+  upper_bound = length * 3
+  hash = Array.new(upper_bound)
 
+  strings.each do |string|
+    hashed_index = get_hashed_index(string, upper_bound)
+    if hash[hashed_index].nil?
+      hash[hashed_index] = [string]
+    else
+      hash[hashed_index].push(string)
+    end
+  end
+
+  return get_clean_hash(hash)
+end
+
+def get_numerical_key(string)
+  return string.bytes.sum
```

Comment:
Can different strings (of different lengths) give the same byte sum?  It's unlikely, but is it impossible... I'm not sure.

Also just note this is an O(m) operation where m is the length of the string.